### DISCUSSION TITLE
Backup container creates volumes, which aren't needed

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#2178] [Questions] [answered]
If I for example have elastic search disabled, run the backup, delete the borg container, then I have an empty Elasticsearch volume. I wouldn't say that this is an Issue, but it is an unnecessary-created volume.

**Comment by szaimen** [comment, marked as answer]
Hi, so the problem is that upon instance restore we cannot know if elasticsearch was formerly enabled before we actually run the restore (and then we cannot add the volume anymore). That is why the elasticsearch volume gets always attached even if it should not be used. 

Also doing it in a different way would potentially run into problems when e.g. rsync then tries to delete or restore to the unmounted or mounted volume so that is why I opted for this solution.
